tests/Unit/Security/Authenticator/ApiTokenAuthenticatorTest.php

```diff
@@ -0,0 +1,147 @@
+    protected ApiToken $unvalidToken;
+
+    protected LocalAccount $user;
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function setUp(): void
+    {
+        parent::setUp();
+
+        // Client mock
+        $this->mockOidcClient = $this->createMock(OidcClientInterface::class);
+
+        // User mock
+        $this->mockUserProvider = $this->createMock(LocalUserProvider::class);
+
+        $this->user = $this->createMock(LocalAccount::class);
+        $this->user->method('getUserIdentifier')->willReturn('userid');
+
+
+        // Native Tokens
+        $trustClient = $this->createMock(TrustedClient::class);
+        $expdate = new \DateTimeImmutable('-30 minutes');
+        $nowdate =  new \DateTimeImmutable('+5 minutes');
+
+        $this->validToken = new ApiToken($this->user, $trustClient, $nowdate);
+        $this->unvalidToken = new ApiToken($this->user, $trustClient, $expdate);
+
+        // Token repo
+        $this->mockTokenRepo = $this->createMock(ApiTokenRepository::class);
+        $tokenMap = [
+            [$this->validToken->token, null, null, $this->validToken],
+            [$this->unvalidToken->token, null, null, $this->unvalidToken],
+        ];
+        $this->mockTokenRepo->method('find')->willReturnMap($tokenMap);
+
+        // Authenticator
+        $this->auth = new ApiTokenAuthenticator(
+            $this->mockTokenRepo
+        );
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function tearDown(): void
+    {
+        parent::tearDown();
+
+        unset($this->auth);
+        unset($this->mockOidcClient);
+        unset($this->mockUserProvider);
+        unset($this->mockTokenRepo);
+    }
+
+    public function testSupports(): void
+    {
+        $request = $this->mockRequest('1234');
+        self::assertTrue($this->auth->supports($request));
+    }
+
+    public function testAuthenticate(): void
+    {
+        //Valid native token
+        $request = $this->mockRequest($this->validToken->token);
+        $passport = $this->auth->authenticate($request);
+        $badge = $passport->getBadge(UserBadge::class);
+        self::assertNotNull($badge);
+        self::assertInstanceOf(UserBadge::class, $badge);
+        self::assertSame($badge->getUserIdentifier(), $this->user->getUserIdentifier());
+
+        //Unvalid native token
+        $request = $this->mockRequest($this->unvalidToken->token);
+        self::expectException(CredentialsExpiredException::class);
+        $this->auth->authenticate($request);
+    }
+
+    private function mockRequest(string $token): MockObject&Request
+    {
+        /** @var MockObject&Request $request */
+        $request = $this->createMock(Request::class);
+        $request->method('getRequestUri')->willReturn('/api/');
+        $request->headers = new HeaderBag([
+            'Authorization' => "Bearer $token",
+        ]);
+        return $request;
+    }
+
+    public function testOnAuthenticationSuccess(): void
```

Comment:
curious to hear what you think, but I think we should strive to fully cover our newly added code.